site/_posts/2023-03-09-react-example-application.md

```diff
@@ -0,0 +1,247 @@
+- Password: `password`
+
+This will log you in as an agent. You can also log in as a user by using the following credentials:
+- Email: `richard@example.com`
+- Password: `password`
+
+### Set up the Fusionauth Instance Manually
+
+If you do not want to use Docker, you can set up the FusionAuth instance manually. To do so, follow the steps detailed in the [FusionAuth documentation](https://fusionauth.io/docs/v1/tech/installation-guide/fusionauth-app) and the [README.md](https://github.com/FusionAuth/fusionauth-react-example-app-placeholder-replaceme/blob/main/README.md) of the example application.
+
+## Using Hosted OAuth Service Provider Endpoints
+
+Authentication is hard. We want to make it easy.
+That is why we have added the necessary endpoints for the OAuth 2.0 login flow into the FusionAuth API. This means that you can use FusionAuth to authenticate users. This is a great way to get started with FusionAuth and to use it as a drop-in replacement for your existing OAuth 2.0 provider. So you can concentrate on providing the best experience for your users.
+
+{% include _callout-important.liquid
+content=
+"For limitations and prerequisites, see the [Hosted OAuth Service Provider Endpoints](https://fusionauth.io/xyz) documentation."
+%}
+
+{% include _image.liquid src="/assets/img/blogs/react-example-application/fusiondesk-fusionauth-login.png" alt="Login Screen of FusionAuth with the styling of FusionDesk" class="img-fluid" figure=false %}
+
+As you can see - if you inspect the backend of the example application - we do not provide any OAuth 2.0 endpoints. Instead, we use the FusionAuth API to authenticate users.
+
+When the user accesses the backend, we validate the `access_token` cookie. If the cookie is valid, we will return the tickets. If the cookie is not valid, we return a 401 error.
+
+See `server/src/middlewares/fusion-auth.middleware.ts` for the implementation.
+
+{% include _callout-note.liquid
+content=
+"Currently every request validates the `access_token` with FusionAuth using the `/oauth2/introspect` endpoint. This is not the most efficient way to validate the token. Depending on the security requirements of your application, you might want to cache this information or validate the `access_token` directly without using the FusionAuth API."
+%}
+
+We also use the FusionAuth API to retrieve user information for the creator of a ticket. This is done in the `TicketController.ts` file using the `retrieveUser` method of the FusionAuth API client for typescript.
+
+### Theming the Login / Register View in Fusionauth
+
+The login / register view in FusionAuth can be themed using the FusionAuth theme editor. You can find more information about the theme editor in the [FusionAuth documentation](https://fusionauth.io/docs/v1/tech/themes/).
+
+For this example application, we integrate Tailwind CSS and DaisyUI into the theme by automatically generating the css stylesheet based on the FusionAuth theme templates.
```

Comment:
I want to know more about this. Can you put in a few more paragraphs about how you automatically generate the CSS from the theme templates?